[MANUNTAG MANUEL II PAPA] iP

manuelmanuntag96 opened the pull request.

manuelmanuntag96 committed:
Level-0: Add Greet

manuelmanuntag96 committed:
Level 1. Greet, Echo, Exit
Implement a skeletal version of Duke that starts by greeting the user, simply echos commands entered by the user, and exits when the user types bye.

manuelmanuntag96 committed:
Level 2. Add, List
Add the ability to store whatever text entered by the user and display them back to the user when requested.

manuelmanuntag96 committed:
Level 3. Mark as Done
Add the ability to mark tasks as done.
Use a class to represent tasks.

manuelmanuntag96 committed:
A-CodingStandard
Tweak the code to comply with a coding standard.

NeilBaner commented on src/main/java/Duke.java:
```diff
@@ -6,5 +8,76 @@ public static void main(String[] args) {
         System.out.println("Hello from\n" + logo);
+
+        String line;
+        String list = "list";
```
Could this be named a bit more descriptively? As it stands I'm not entirely sure what this variable is for.

NeilBaner commented on src/main/java/Duke.java:
```diff
@@ -6,5 +8,76 @@ public static void main(String[] args) {
+
+        String line;
+        String list = "list";
+        String bleh = "bleh";
```
Ditto for here, perhaps the name could be more descriptive about the function/use of the variable rather than the content?

NeilBaner commented on src/main/java/Duke.java:
```diff
@@ -6,5 +8,76 @@ public static void main(String[] args) {
+        String line;
+        String list = "list";
+        String bleh = "bleh";
+        String bye = "bye";
```
Same as above.

NeilBaner commented on src/main/java/Duke.java:
```diff
@@ -6,5 +8,76 @@ public static void main(String[] args) {
+        Task[] taskList = new Task[100];
+        int i = 0;
+
+        System.out.println("____________________________________________________________");
```
Might it be a good idea to extract this out as a constant so you call that, instead of repeating this line of code over and over?

NeilBaner commented on src/main/java/Duke.java:
```diff
@@ -6,5 +8,76 @@ public static void main(String[] args) {
+                System.out.println("____________________________________________________________");
+                int number = 1;
+                System.out.println(" Here are the tasks in your list:");
+                while (number <= i) {
```
Would a for loop be appropriate here?

NeilBaner commented on src/main/java/Duke.java:
```diff
@@ -6,5 +8,76 @@ public static void main(String[] args) {
+            } else if(line.contains("done")){
+                int dividerPosition = line.indexOf(" ");
+                String taskNumber = line.substring(dividerPosition+1);
+                int number = Integer.parseInt(taskNumber);
```
Perhaps this variable could be named more clearly so it's more obvious what it's for?

NeilBaner commented on src/main/java/Duke.java:
```diff
@@ -6,5 +8,76 @@ public static void main(String[] args) {
+            return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+        }
+
+        public static void markAsDone(Task t) {
```
Not sure if this is a violation of any standards per se, but would this method be better as a non-static method that modifies the specific instance? It might clean up some of the code where you call it too. Is there any reason this is static?

NeilBaner commented on src/main/java/Duke.java:
```diff
@@ -6,5 +8,76 @@ public static void main(String[] args) {
+        }
+
+        public String getStatusIcon() {
+            return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
```
I like this, an ingenius solution in my opinion.

NeilBaner commented on src/main/java/Duke.java:
```diff
@@ -6,5 +8,76 @@ public static void main(String[] args) {
+    }
+
+    public static class Task {
+        protected String description;
```
Not sure if this is a standards violation, but OOP best practice as far as I am aware is to grant as little access as necessary. Should this perhaps be private instead of protected?

manuelmanuntag96 committed:
Level 4. ToDos, Events, Deadlines
Add support for tracking three types of tasks:

GoldenCorgi commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+
+    public static void taskList(Task[] taskList, int taskCount) {
+        for(int i=0; i<taskCount;i++) {
+            if(taskList[i].taskType == "T") {
```
Perhaps it is good practice to use taskList[i].taskType.equals("T") for string comparison here?

GoldenCorgi reviewed (commented):
Otherwise, looks good to me!

WYing333 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+    public static int addTask(Task[] taskList, int taskCount, String taskName, String taskBy) {
+        taskList[taskCount] = new Deadline(taskName, taskBy);
+
+        System.out.println(" Got it. I've added this task:\n" + "  [" + taskList[taskCount].taskType + "]["+ taskList[taskCount].getStatusIcon() + "] " + taskList[taskCount].description + " (" + taskList[taskCount].by + ")"  + "\n" + " Now you have " + (taskCount+1) + " tasks in the list.");
```
This statement is quite long, maybe could use some refactors to make it shorter.

WYing333 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,77 @@
+        do{
+            line = in.nextLine();
+            if(line.contains("list")) {
+                System.out.println("____________________________________________________________");
```
Maybe this could be refactored with some other print statements to a function so that do not need to write this long statement everytime.

WYing333 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,38 @@
+        this.taskType = "T";
+    }
+
+    public static int addTask(Task[] taskList, int taskCount, String taskName) {
+        taskList[taskCount] = new Task(taskName);
+
+        System.out.println(" added: " + taskList[taskCount].description);
+
+        taskCount++;
+        return taskCount;
+    }
+
+    public static void taskList(Task[] taskList, int taskCount) {
+        for(int i=0; i<taskCount;i++) {
+            System.out.println(" " + (i+1) + ".[" + taskList[i].taskType + "][" + taskList[i].getStatusIcon() + "] " + taskList[i].description);
+        }
+    }
+
+    public static void markAsDone(Task[] taskList, int taskCount, int taskNum) {
+        taskList[taskNum-1].isDone = true;
+        System.out.println(" Nice! I've marked this task as done:\n" + "   [" + taskList[taskNum-1].getStatusIcon() + "] " + taskList[taskNum-1].description );
+    }
```
I think these methods fullfill the requirement quite well.

manuelmanuntag96 committed:
A-TextUiTesting
Test using the I/O redirection technique

manuelmanuntag96 committed:
A-CodeQuality
Improve code quality

manuelmanuntag96 committed:
Level-5-001
Fix todo error when description is empty

manuelmanuntag96 committed:
Level-5-002
Fixed error for when Event and Deadline descriptor is empty.

manuelmanuntag96 committed:
Level-5-003
add DukeException for when deadline by and event @ is missing

manuelmanuntag96 committed:
Level-5-004
add a catch for NullPointerException for when done number is invalid

manuelmanuntag96 committed:
Merge branch 'branch-Level-5'

manuelmanuntag96 committed:
A-Packages
Divide classes into packages

manuelmanuntag96 committed:
A-Exceptions v2
add error handling for invalid inputs

manuelmanuntag96 committed:
Level 5.1: Turn Array to ArrayList

manuelmanuntag96 committed:
Level 6. Delete
Add support for deleting tasks from the list.

manuelmanuntag96 committed:
Level 7. Save
Save the tasks in the hard disk automatically whenever the task list changes.
Load the data from the hard disk when Duke starts up.
Hard-code the file name and location

manuelmanuntag96 committed:
Ignore duke.txt

lowjiayee commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+        } else {
+            taskList.add(new Deadline(taskName, taskBy));
+
+            System.out.println(" Got it. I've added this task:\n" + "  [" + ((Task)taskList.get(taskCount)).taskType + "]["+ ((Task)taskList.get(taskCount)).getStatusIcon() + "] " + ((Task)taskList.get(taskCount)).description + " (" + ((Task)taskList.get(taskCount)).by + ")"  + "\n" + " Now you have " + (taskCount+1) + " tasks in the list.");
```
Perhaps can break this into multiple lines to meet coding standard?

lowjiayee commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+            if(((Task)taskList.get(i)).taskType == "T") {
+                System.out.println(" " + (i + 1) + ".[" + ((Task)taskList.get(i)).taskType + "][" + ((Task)taskList.get(i)).getStatusIcon() + "] " + ((Task)taskList.get(i)).description);
+            } else {
+                System.out.println(" " + (i + 1) + ".[" + ((Task)taskList.get(i)).taskType + "][" + ((Task)taskList.get(i)).getStatusIcon() + "] " + ((Task)taskList.get(i)).description + " (" + ((Task)taskList.get(i)).by + ")");
```
Perhaps can break this into multiple lines to meet coding standard?

lowjiayee commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,131 @@
+import java.nio.file.Paths;
+
+public class Duke {
+    public static final int MAX_SIZE = 100;
```
Perhaps can consider removing this, since this is unused?

lowjiayee commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+            }
+        }
+    }
+
+
+
```
Can remove extra lines.

lowjiayee commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,131 @@
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("☹ OOPS!!! The number cannot be out of bounds.");
+                }
+            } else if (!line.contains("bye") & !line.contains("todo") & !line.contains("event") & !line.contains("deadline") & !line.contains("list") & !line.contains("done") & !line.contains("delete")){
```
Can use `default` for this if you switch to `switch..case`

lowjiayee commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,131 @@
+
+        do{
+            line = in.nextLine();
+            if(line.contains("list")) {
```
Can use `case "list":` if you use `switch..case, then adjust accordingly for other commands.

lowjiayee commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,28 @@
+        } else {
+            taskList.add(new Event(taskName, taskAt));
+
+            System.out.println(" Got it. I've added this task:\n" + "  [" + ((Task)taskList.get(taskCount)).taskType + "]["+ ((Task)taskList.get(taskCount)).getStatusIcon() + "] " + ((Task)taskList.get(taskCount)).description + " (" + ((Task)taskList.get(taskCount)).by + ")"  + "\n" + " Now you have " + (taskCount+1) + " tasks in the list.");
```
Perhaps can break this into multiple lines to meet coding standard?

lowjiayee commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,111 @@
+        System.out.println(" added: " + ((Task)taskList.get(taskCount)).description);
+
+
+
```
You may consider removing unnecessary lines for better readability.

lowjiayee commented on src/main/java/duke/WriteToFile.java:
```diff
@@ -0,0 +1,21 @@
+public class WriteToFile {
+    public static void main(ArrayList taskList) {
+        try {
+            FileWriter myWriter = new FileWriter("D:\\Documents\\Modules\\Sem 03\\CS2113\\iP\\data\\duke.txt");
```
Might not work on other systems with no D drive.
This won't work in non-Windows machine because of different path naming convention.
You may consider using relative path instead of absolute path.

lowjiayee commented on src/main/java/duke/CreateFileDir.java:
```diff
@@ -0,0 +1,24 @@
+public class CreateFileDir {
+    public static void CreateFile() {
+        try {
+            Files.createDirectories(Paths.get("D:\\Documents\\Modules\\Sem 03\\CS2113\\iP\\data"));
+            File myObj = new File("D:\\Documents\\Modules\\Sem 03\\CS2113\\iP\\data\\duke.txt"); //D:\Documents\Modules\Sem 03\CS2113\iP\data
```
Might not work on other systems with no D drive.
This won't work in non-Windows machine because of different path naming convention.
You may consider using relative path instead of absolute path.

lowjiayee replied:
Perhaps use a better variable name for `myObj`?

lowjiayee reviewed (commented):
There are other places in your code that you can consider breaking into multiple lines to meet coding standard.

manuelmanuntag96 committed:
Level 9. Find

Give users a way to find a task by searching for a keyword.

manuelmanuntag96 committed:
Level 8. Dates and Times

Teach Duke how to understand dates and times.

manuelmanuntag96 committed:
A-MoreOOP
Make the code more OOP

Refactor the code to extract out closely related code as classes.

manuelmanuntag96 committed:
A-JavaDoc

Add JavaDoc comments

manuelmanuntag96 committed:
Revert "A-JavaDoc"

This reverts commit 11dc4be9fd2b2981eb3feb439df2662e20116b06.

manuelmanuntag96 committed:
Revert "Revert "A-JavaDoc""

This reverts commit b82a45a8cf13bdc7ccaa6c639a2af3f2270c6cef.

manuelmanuntag96 committed:
A-UserGuide

Add a User Guide to the project

manuelmanuntag96 committed:
Fixed the User Guide

manuelmanuntag96 committed:
Revert "Fixed the User Guide"

This reverts commit a6fd5f6412a98f152e8745d17f4035d7078fe8f8.

manuelmanuntag96 committed:
Fixed the User Guide
